Firestore/Source/API/FSTUserDataConverter.mm

```diff
@@ -204,16 +216,22 @@ - (const FieldPath *)path;
 
 - (void)appendToFieldMaskWithFieldPath:(FieldPath)fieldPath;
 
+- (const std::vector<FieldTransform> *)fieldTransforms;
+
+- (void)appendToFieldTransformsWithFieldPath:(FieldPath)fieldPath
+                          transformOperation:
+                              (std::unique_ptr<TransformOperation>)transformOperation;
 @end
 
 @implementation FSTParseContext {
   /** The current path being parsed. */
   // TODO(b/34871131): path should never be nullptr, but we don't support array paths right now.
   std::unique_ptr<FieldPath> _path;
-  // _fieldMask is shared across all active context objects to accumulate the result. For example,
-  // the result of calling any of contextForField, contextForFieldPath and contextForArrayIndex
-  // shares the ownership of _fieldMask.
+  // _fieldMask and _fieldTransforms are shared across all active context objects to accumulate the
```

Comment:
I guess alternatively, a single context object could be wrapped in a `shared_ptr`. Can you please provide some context on how it was implemented in Objective-C?